### PR TITLE
Fix: [CMake] Don't include regression AIs in bundles

### PR DESCRIPTION
**Pull request by glx22**
## Motivation / Problem
In #8906, I though it was a smart move to install script compat files from build dir. But I was forgetting regression AIs were also in this directory. And they ended up in bundles.

## Description
This is a revert of the install related changes I made in #8906.
I then excluded any `CMakeLists.txt` from installation.

I tried another option, using build dir as install source, but the regex to filter AIs subdirs was way harder to read (I'll show it in a suggestion comment).

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: cmake/InstallAndPackage.cmake [hunk 1]
```diff
@@ -26,11 +26,13 @@ install(TARGETS openttd
 install(DIRECTORY
                 ${CMAKE_BINARY_DIR}/lang
                 ${CMAKE_BINARY_DIR}/baseset
-                ${CMAKE_BINARY_DIR}/ai
-                ${CMAKE_BINARY_DIR}/game
+                ${CMAKE_SOURCE_DIR}/bin/ai
+                ${CMAKE_SOURCE_DIR}/bin/game
```
Comment (glx22) [inline cmake/InstallAndPackage.cmake]:
```suggestion
                ${CMAKE_BINARY_DIR}/ai
                ${CMAKE_BINARY_DIR}/game
```

File: cmake/InstallAndPackage.cmake [hunk 2]
```diff
@@ -26,11 +26,13 @@ install(TARGETS openttd
...
                 ${CMAKE_SOURCE_DIR}/bin/scripts
         DESTINATION ${DATA_DESTINATION_DIR}
-        COMPONENT language_files)
+        COMPONENT language_files
+        PATTERN "CMakeLists.txt" EXCLUDE
```
Comment (glx22) [inline cmake/InstallAndPackage.cmake]:
```suggestion
        REGEX "ai/[^\.]+$" EXCLUDE
```

**Review by glx22 — COMMENTED** [review]
Suggested changes are the other possible solution.

I personally prefer this version as it gets a clean install of `ai` and `game` dirs from an unclean checkout.

**Comment by LordAro** [comment]
Which version do you mean by "this version" ? I don't particularly mind either way, though working well from an unclean checkout is a positive

**Comment by glx22** [comment]
"this version" is the install from build dir with a regex to remove regression AI :)

**Review by LordAro — APPROVED** [review]

**Pull request merged**